### PR TITLE
Added client libraries for Ground

### PR DESCRIPTION
**Pull request by andreaskari**
Created client libraries in Python and Java for Ground

### CODE REVIEW

File: client/java/.gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+*.class
```
Comment (vsreekanti) [inline client/java/.gitignore]: Please remove nested `.gitignore`s. If there's something that needs to be ignore, it should be in the toplevel `.gitignore`.

File: build.sbt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import de.johoop.jacoco4sbt.XMLReport
 
-name := """ground"""
+name := """ground"""
```
Comment (vsreekanti) [inline build.sbt]: Whitespace.

File: client/java/.swagger-codegen-ignore [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Swagger Codegen Ignore
```
Comment (vsreekanti) [inline client/java/.swagger-codegen-ignore]: This seems to be empty? If unnecessary, please remove.

File: client/java/.travis.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#
```
Comment (vsreekanti) [inline client/java/.travis.yml]: There's no need for a special `.travis.yml` for each client. The necessary logic here should be merged into the top level `.travis.yml`.

File: client/java/README.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+Refer to the [official documentation](https://maven.apache.org/plugins/maven-deploy-plugin/usage.html) for more information.
+
+### Maven users
+
+Add this dependency to your project's POM:
+
+```xml
+<dependency>
+    <groupId>io.swagger</groupId>
+    <artifactId>swagger-java-client</artifactId>
```
Comment (vsreekanti) [inline client/java/README.md]: Shouldn't this reflect the actual project and be `edu.berkeley.ground.ground-java-client`? We wouldn't be able to publish this to Maven central under `io.swagger`, and the examples below seem like dummy examples.

File: client/java/README.md [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+*DefaultApi* | [**versionsLineageEdgesPost**](docs/DefaultApi.md#versionsLineageEdgesPost) | **POST** /versions/lineage_edges | 
+*DefaultApi* | [**versionsLineageGraphsIdGet**](docs/DefaultApi.md#versionsLineageGraphsIdGet) | **GET** /versions/lineage_graphs/{id} | 
+*DefaultApi* | [**versionsLineageGraphsPost**](docs/DefaultApi.md#versionsLineageGraphsPost) | **POST** /versions/lineage_graphs | 
+*DefaultApi* | [**versionsNodesIdGet**](docs/DefaultApi.md#versionsNodesIdGet) | **GET** /versions/nodes/{id} | 
+*DefaultApi* | [**versionsNodesPost**](docs/DefaultApi.md#versionsNodesPost) | **POST** /versions/nodes | 
+*DefaultApi* | [**versionsStructuresIdGet**](docs/DefaultApi.md#versionsStructuresIdGet) | **GET** /versions/structures/{id} | 
+*DefaultApi* | [**versionsStructuresPost**](docs/DefaultApi.md#versionsStructuresPost) | **POST** /versions/structures | 
+
+
+## Documentation for Models
```
Comment (vsreekanti) [inline client/java/README.md]: Remove unnecessary headings.

File: client/java/build.gradle [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+    android {
+        compileSdkVersion 25
+        buildToolsVersion '25.0.2'
+        defaultConfig {
+            minSdkVersion 14
+            targetSdkVersion 25
+        }
+        compileOptions {
+            sourceCompatibility JavaVersion.VERSION_1_7
```
Comment (vsreekanti) [inline client/java/build.gradle]: Ground uses Java 8. The client should also.

File: client/java/build.sbt [hunk 8]
```diff
@@ -0,0 +1,20 @@
+lazy val root = (project in file(".")).
```
Comment (vsreekanti) [inline client/java/build.sbt:1]: Why is there both a `build.sbt` and a `build.gradle` in the Java client?

File: client/java/git_push.sh [hunk 9]
```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
```
Comment (vsreekanti) [inline client/java/git_push.sh]: Why do we need this file?

File: client/java/settings.gradle [hunk 10]
```diff
@@ -0,0 +1 @@
+rootProject.name = "swagger-java-client"
```
Comment (vsreekanti) [inline client/java/settings.gradle]: Again, we should have a non-generic project name. Same with the `pom.xml` above.

File: client/java/src/main/java/io/swagger/client/ApiCallback.java [hunk 11]
```diff
@@ -0,0 +1,62 @@
+/*
```
Comment (vsreekanti) [inline client/java/src/main/java/io/swagger/client/ApiCallback.java]: It seems like everything has been generated with the generic template. Things should be in the `edu.berkeley.ground.client` package.

File: client/python/.gitignore [hunk 12]
```diff
@@ -0,0 +1,64 @@
+# Byte-compiled / optimized / DLL files
```
Comment (vsreekanti) [inline client/python/.gitignore]: Same comments for the Python skeleton.

File: client/python/docs/DefaultApi.md [hunk 13]
```diff
@@ -0,0 +1,1081 @@
+# swagger_client.DefaultApi
```
Comment (vsreekanti) [inline client/python/docs/DefaultApi.md]: Can we avoid duplication of this? Seems like the same as the Java version.

File: client/python/setup.py [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+    OpenAPI spec version: 
+    Contact: andreaskari@berkeley.edu
+    Generated by: https://github.com/swagger-api/swagger-codegen.git
+"""
+
+
+import sys
+from setuptools import setup, find_packages
+
+NAME = "swagger-client"
```
Comment (vsreekanti) [inline client/python/setup.py]: Same comment about naming.

**Review by vsreekanti — CHANGES_REQUESTED** [review]
A few other points: 

* Why is `gradle-wrapper.jar` being committed? 
* Let's turn off the contact information field in the autogenerated code. Doesn't make sense to put an email there when we want people to engage via Github.
* If we're not going to have tests, let's remove that frm the skeleton as well.

**Comment by andreaskari** [comment]
The latest commit on "master" and "client" branches have the latest and greatest clients. I did a major cleanup. Let me know what else must be done.

**Comment by vsreekanti** [comment]
Looks much better, thanks! Unfortunately, tests aren't passing. You should be able to look more closely [here](https://travis-ci.org/ground-context/ground/builds/265373431?utm_source=github_status&utm_medium=notification).

One nit: I'm still not sure why we have multiple build files (Maven, SBT, Gradle) in the Java project.

Once the tests pass, we should be good to go with this!

**Comment by andreaskari** [comment]
All the tests are now passing and everything should be good to go

**Comment by vsreekanti** [comment]
I might have missed something, @andreaskari, but it seems like the reason the tests are passing is because we aren't actually testing the client libraries. When I said earlier I wanted the tests to pass, I meant that we should be testing the client libraries like you were earlier, but we should modify the environment so that the client library tests are passing in addition to the server-side tests. Let me know if you need any help with this.

Also, there have been some changes to the master branch since you opened the PR, so please merge the latest changes into your branch.

**Review by vsreekanti — APPROVED** [review]

**Pull request merged**